### DISCUSSION TITLE
Decoder layer input/output channels

### DISCUSSION THREAD

**Discussion by ckeen2019** [milesial/Pytorch-UNet#281] [Q&A]
Hello,

I noticed that you implemented your decoder (Up) layers in a way that used 1024, 512, etc as the input channels as opposed to 1024 + 512, 512 + 256, etc, and then output half the number of channels as expected so that the concatenation in the next layer adds to 1024, 512, etc channels. Is there any advantage to this method? I've seen some other implementations use the other method I described and I'm not sure if one is better than the other.

I also saw that your implementation (assuming bilinear) never actually goes up to 1024 channels. In your bottom level, you use 512 channels in every operation. Why not add another class to unet_parts.py like the example below?

```
class Bottom(nn.Module):
  
  def __init__(self, in_channels, out_channels):
    super().__init__()
    self.bottom_conv = nn.Sequential(
      nn.MaxPool2d(2),
      DoubleConv(in_channels, out_channels, in_channels * 2)
    )

  def forward(self, t):
    return self.bottom_conv(t) 
```

Thanks
